### PR TITLE
Optional: improve readme example

### PR DESCRIPTION
**Pull request by vict0rsch**
1. I think having the imports is part of the tutorials: it helps ppl understand the package's structure
2. I added a `tqdm` progress bar because running a silent example is always confusing (not sure it's working)
3. `if __name__ == "__main__":` is optional but I think it's good to spread good practice
4. Same for applying `black` to the code example
5. I proposed a title, optional again
6. I did some invisible Markdown syntax formatting (adding spaces around titles or lists for instance)
7. Some trailing whitespace removal happened automatically with [Trailing Spaces](https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces) but it does not hurt :p

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,70 +13,97 @@ pip install -e .
...
 ```bash
-python train.py --env HyperGrid --env.ndim 4 --env.height 8 --n_iterations 100000 --loss TB 
+python train.py --env HyperGrid --env.ndim 4 --env.height 8 --n_iterations 100000 --loss TB
 python train.py --env DiscreteEBM --env.ndim 4 --env.alpha 0.5 --n_iterations 10000 --batch_size 64 --temperature 2.
-python train.py --env HyperGrid --env.ndim 2 --env.height 64 --n_iterations 100000 --loss DB --replay_buffer_size 1000 --logit_PB.module_name Uniform --optim sgd --optim.lr 5e-3 
+python train.py --env HyperGrid --env.ndim 2 --env.height 64 --n_iterations 100000 --loss DB --replay_buffer_size 1000 --logit_PB.module_name Uniform --optim sgd --optim.lr 5e-3
 python train.py --env HyperGrid --env.ndim 4 --env.height 8 --env.R0 0.01 --loss FM --optim adam --optim.lr 1e-4
 ```
 
 ### Example, in a few lines
```
Comment (saleml) [inline README.md:36]: can you add a line saying you need to pip install tqdm for this ? It's not in `setup.py` (but it is in `requirements.txt`)

Comment (vict0rsch) [reply README.md:36]: According to this discussion: https://stackoverflow.com/a/33685899/3867406

Do you want tqdm to be core to your package (add to `setup.py`) or more of a deployment req (`requirements.txt`)?

Comment (saleml) [reply README.md:36]: It shouldn't be core to the package. tqdm, wandb, etc... are in `requirements.txt` exactly because of this reason. They're helpful to run the accompanying scripts, but are not necessary to define / train a gflownet.

**Review by saleml — APPROVED** [review]
LGTM
